cookbooks/aws-parallelcluster-slurm/files/default/config_slurm/scripts/health_checks/gpu_health_check.sh

```diff
@@ -0,0 +1,209 @@
+
+  ## Retrieve gpus information and enable them to run the health check
+  local gpus_list
+  gpus_list=${GPU_DEVICE_ORDINAL//,/$'\n'}
+  for gpu_id in $gpus_list
+  do
+    # Retrieve gpu information. This check will exit in following cases:
+    # - the nvidia-smi command is not available
+    # - the nvidia-smi command is available but it fails its execution
+    nvidia_smi_out=$(nvidia-smi -q -i $gpu_id)
+    nvidia_smi_exit_code=$(echo $?)
+    if [ $nvidia_smi_exit_code -eq 127 ]; then
+      fast_success_exit $nvidia_smi_exit_code "Health check has been executed but the nvidia-smi tool is not available in this system or is located in a different file path. If possible use an official Parallel Cluster AMI."
+    fi
+    fast_success_exit $nvidia_smi_exit_code "The nvidia-smi tool fails its execution."
+
+    ## Enable persistence mode in target GPU, required by the DCGM Diagnostic tool
+    log_info "Enabling persistence mode in target gpu {$gpu_id} if not already enabled"
+    local persistence_mode_enabled_$gpu_id
+    persistence_mode=$(echo "$nvidia_smi_out" | grep "Persistence Mode")
+    if [[ ! ${persistence_mode} == *"Enabled"* ]]; then
+      sudo nvidia-smi -i $gpu_id -pm 1
+      declare persistence_mode_enabled_$gpu_id=true
+    fi
+
+    ## Log useful GPU information
+    log_info "GPU : $gpu_id"
+    log_info "$(echo "$nvidia_smi_out" | grep "GPU UUID")"
+    log_info "$(echo "$nvidia_smi_out" | grep "Serial Number")"
+  done
+
+  ## Run GPUs Health Check test
+  log_info "Running GPU Health Check tool"
+  dcgmi diag -i $GPU_DEVICE_ORDINAL -r 2
+  dcgmi_exit_code=$(echo $?)
+
+  if [ $dcgmi_exit_code -ne 0 ]; then
+    if [ $dcgmi_exit_code -eq 127 ]; then
+      log_info "Health check has been executed but the DCGM Diagnostic tool is not available in this system or is located in a different file path. If possible use an official Parallel Cluster AMI."
+      dcgmi_exit_code=0
+    fi
+  else
+    log_info "The GPU Health Check succeeded"
+  fi
+  tear_down
+  return $dcgmi_exit_code
+}
+
+# FIXME REMOVE
+function install_dcgmi() {
+  # https://docs.nvidia.com/datacenter/dcgm/latest/user-guide/getting-started.html
+  # https://developer.download.nvidia.com/compute/cuda/repos/
+  # Alinux2
+  cd /etc/yum.repos.d/
+  sudo wget https://developer.download.nvidia.com/compute/cuda/repos/rhel7/x86_64/cuda-rhel7.repo
+  sudo yum install -y datacenter-gpu-manager
+}
+# END FIXME REMOVE
+
+function _log() {
+  echo "$(date +"%Y-%m-%d %H-%M-%S,%3N") - [${0##*/}] - $1 - JobID $SLURM_JOB_ID - " "${@:2}"
+}
+
+function log_info() {
+  _log "INFO" "$@"
+}
+
+function log_error() {
+  _log "ERROR" "$@"
+}
+
+function fast_success_exit() {
+  if [ $1 -ne 0 ]; then
+    log_info "$2"
+    tear_down
+    exit 0
+  fi
+}
+
+function is_nvidia()
+{
+  # Instances from https://aws.amazon.com/ec2/instance-types/
+  case "${instance_type}" in
+    g2.*|g3.*|g3s.*|g4dn.*|g5.*|g5g.*|p2.*|p3.*|p3dn.*|p4d.*|p4de.*) return ;;
```

Comment:
we shouldn't have the list hardcoded.
we can look at /opt/parallelcluster/shared/instance-types-data.json which contains the describe instances output, to check if instance has GPU info, e.g.

```
    "GpuInfo": {
      "Gpus": [
        {
          "Name": "T4",
          "Manufacturer": "NVIDIA",
          "Count": 4,
          "MemoryInfo": {
            "SizeInMiB": 16384
          }
        }
      ],
      "TotalGpuMemoryInMiB": 65536
    },
```

Then we can remove also the call to the metadata service